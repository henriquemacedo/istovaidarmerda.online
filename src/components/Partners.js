import React from "react";
import PropTypes from "prop-types";
import Link from "next/link";
import styled from "styled-components";
import { rem } from "polished";

const Wrapper = styled.section`
  background-color: var(--yellow);
  padding: 100px;
  text-align: center;
`;

const Logos = styled.div`
  --itemWidth: 250px;
  width: min-content;
  margin: 50px auto 0 auto;

  ul {
    margin: 0;
    display: grid;
    grid-template-columns: repeat(4, var(--itemWidth));
    grid-gap: 50px;
    padding: 0;
  }
`;

const Logo = styled.li`
  list-style: none;
  position: relative;
  border-radius: ${rem("50px")};
  background-color: ${({ bg }) => bg || "var(--light)"};
  width: var(--itemWidth);
  height: var(--itemWidth);
  box-shadow: 0 0 ${rem("50px")} rgba(33, 33, 33, 0.25);
  transition-duration: 0.5s;

  a {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;

    img {
      max-width: 180px;
      max-height: 150px;
    }
  }

  &:hover {
    box-shadow: 0 0 ${rem("75px")} rgba(33, 33, 33, 0.5);
  }
`;

const Partner = ({ bg, logo, name, url }) => (
  <Logo bg={bg}>
    <Link href={url} passHref>
      <a target="_blank">
        <img src={logo} alt={name} />
      </a>
    </Link>
  </Logo>
);

const Partners = ({ partners }) => (
  <Wrapper>
    <h2>Apoios</h2>
    <Logos>
      <ul>
        {Array.isArray(partners) &&
          partners.map((partner, index) => {
            const key = `partner-${index}`;
            return <Partner key={key} bg={partner.bg} logo={partner.logo} name={partner.name} url={partner.url} />;
          })}
      </ul>
    </Logos>
  </Wrapper>
);

Partner.defaultProps = {
  bg: "#ffffff",
};

Partner.propTypes = {
  bg: PropTypes.string,
  logo: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

Partners.propTypes = {
  partners: PropTypes.arrayOf(
    PropTypes.shape({
      bg: PropTypes.string,
      logo: PropTypes.string,
      name: PropTypes.string,
      url: PropTypes.string,
    }),
  ).isRequired,
};

export default Partners;
