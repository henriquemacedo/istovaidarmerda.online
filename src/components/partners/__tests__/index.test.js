import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Partners from "../index";
import PARTNERS_PROPS from "../mocks/partners";
import Partner from "../partner";

describe("components/partners", () => {
  /**
   * @type {IPartnersProps}
   */
  let props;

  beforeEach(() => {
    props = {
      partners: PARTNERS_PROPS,
    };
  });
  describe("<Partners />r", () => {
    it("should render without errors", () => {
      const component = render(<Partners partners={props.partners} />);

      expect(component).toMatchSnapshot();
    });
  });

  describe("<Partner />r", () => {
    it("should render without errors", () => {
      const data = props.partners[1];
      const component = render(<Partner bg={data.bg} logo={data.logo} name={data.name} url={data.url} />);

      expect(component).toMatchSnapshot();
    });
  });
});
