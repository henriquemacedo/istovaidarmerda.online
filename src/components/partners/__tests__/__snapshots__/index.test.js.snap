// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/partners <Partner />r should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="styles__Logo-sc-12ppjjf-2 WNIod partner"
      >
        <a
          href="https://nacional2podcast.pt"
          rel="noreferrer noopener"
          target="_blank"
        >
          <img
            alt="Nacional 2 Podcast"
            height="175"
            loading="lazy"
            src="/images/partners/n2-podcast.svg"
            width="175"
          />
        </a>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="styles__Logo-sc-12ppjjf-2 WNIod partner"
    >
      <a
        href="https://nacional2podcast.pt"
        rel="noreferrer noopener"
        target="_blank"
      >
        <img
          alt="Nacional 2 Podcast"
          height="175"
          loading="lazy"
          src="/images/partners/n2-podcast.svg"
          width="175"
        />
      </a>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/partners <Partners />r should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="styles__Wrapper-sc-12ppjjf-0 dZVNnC"
      >
        <h2>
          Apoios
        </h2>
        <div
          class="styles__Logos-sc-12ppjjf-1 frbQMY"
        >
          <ul>
            <li
              class="styles__Logo-sc-12ppjjf-2 gQPSUp partner"
            >
              <a
                href="https://inglesparamarketing.com"
                rel="noreferrer noopener"
                target="_blank"
              >
                <img
                  alt="market[ING]"
                  height="175"
                  loading="lazy"
                  src="/images/partners/marketing-logo.svg"
                  width="175"
                />
              </a>
            </li>
            <li
              class="styles__Logo-sc-12ppjjf-2 WNIod partner"
            >
              <a
                href="https://nacional2podcast.pt"
                rel="noreferrer noopener"
                target="_blank"
              >
                <img
                  alt="Nacional 2 Podcast"
                  height="175"
                  loading="lazy"
                  src="/images/partners/n2-podcast.svg"
                  width="175"
                />
              </a>
            </li>
            <li
              class="styles__Logo-sc-12ppjjf-2 btDIZZ partner"
            >
              <a
                href="https://istuff.pt"
                rel="noreferrer noopener"
                target="_blank"
              >
                <img
                  alt="iStuff"
                  height="175"
                  loading="lazy"
                  src="/images/partners/istuff.png"
                  width="175"
                />
              </a>
            </li>
            <li
              class="styles__Logo-sc-12ppjjf-2 WNIod partner"
            >
              <a
                href="https://facebook.com/doccmedia"
                rel="noreferrer noopener"
                target="_blank"
              >
                <img
                  alt="Docc Media"
                  height="175"
                  loading="lazy"
                  src="/images/partners/doccmedia.png"
                  width="175"
                />
              </a>
            </li>
          </ul>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="styles__Wrapper-sc-12ppjjf-0 dZVNnC"
    >
      <h2>
        Apoios
      </h2>
      <div
        class="styles__Logos-sc-12ppjjf-1 frbQMY"
      >
        <ul>
          <li
            class="styles__Logo-sc-12ppjjf-2 gQPSUp partner"
          >
            <a
              href="https://inglesparamarketing.com"
              rel="noreferrer noopener"
              target="_blank"
            >
              <img
                alt="market[ING]"
                height="175"
                loading="lazy"
                src="/images/partners/marketing-logo.svg"
                width="175"
              />
            </a>
          </li>
          <li
            class="styles__Logo-sc-12ppjjf-2 WNIod partner"
          >
            <a
              href="https://nacional2podcast.pt"
              rel="noreferrer noopener"
              target="_blank"
            >
              <img
                alt="Nacional 2 Podcast"
                height="175"
                loading="lazy"
                src="/images/partners/n2-podcast.svg"
                width="175"
              />
            </a>
          </li>
          <li
            class="styles__Logo-sc-12ppjjf-2 btDIZZ partner"
          >
            <a
              href="https://istuff.pt"
              rel="noreferrer noopener"
              target="_blank"
            >
              <img
                alt="iStuff"
                height="175"
                loading="lazy"
                src="/images/partners/istuff.png"
                width="175"
              />
            </a>
          </li>
          <li
            class="styles__Logo-sc-12ppjjf-2 WNIod partner"
          >
            <a
              href="https://facebook.com/doccmedia"
              rel="noreferrer noopener"
              target="_blank"
            >
              <img
                alt="Docc Media"
                height="175"
                loading="lazy"
                src="/images/partners/doccmedia.png"
                width="175"
              />
            </a>
          </li>
        </ul>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
